### PR TITLE
Add x-commit-hash field to the opam file when releasing

### PR DESCRIPTION
**Pull request by gpetiot**
First attempt at fixing #222

### CODE REVIEW

File: tests/test_opam_file.ml [hunk 1]
```diff
@@ -27,7 +31,8 @@ description: "This package is nice"
 |}
   in
   let expected_v1 =
-    {|opam-version: "2.0"
+    {|x-commit-hash: "6814f8b26946358c72b926706f210025f36619b0"
```
Comment (NathanReb) [inline tests/test_opam_file.ml:34]: I'm surprised it's printed as the first field!

Comment (gpetiot) [reply tests/test_opam_file.ml:34]: I looked this up, it does'nt look like we have much control upon this, we are just calling write_to_string

Comment (NathanReb) [reply tests/test_opam_file.ml]: Yeah I wasn't expecting we'd have control over this. It doesn't matter much anyway!

**Comment by NathanReb** [comment]
@gpetiot any news on this front? Is it still a draft?

**Comment by gpetiot** [comment]
I haven't look at this in a while, I think I remember the e2e test was incomplete (the opam file tested is not the right one), plus testing with different tag settings is not done either.

File: tests/bin/x-commit-hash/run.t [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+We create an opam package:
+
+    $ dune-release opam pkg --dry-run
+    [-] Creating opam package description for whatever
+    dune-release: [WARNING] Could not find _build/whatever-0.1.0.url.
+    dune-release: [WARNING] using https://github.com/foo/whatever/releases/download/0.1.0/whatever-0.1.0.tbz for as url.src. Note that it might differ from the one generated by Github
+    [+] Wrote opam package description _build/whatever.0.1.0/opam
+    $ cat _build/whatever.0.1.0/opam
+    cat: _build/whatever.0.1.0/opam: No such file or directory
```
Comment (gpetiot) [inline tests/bin/x-commit-hash/run.t:98]: That's weird, this file should exist and contain the x-commit-hash field

Comment (NathanReb) [reply tests/bin/x-commit-hash/run.t:98]: Hmm yes indeed, any idea what's wrong here? Could you look into it?

Comment (NathanReb) [reply tests/bin/x-commit-hash/run.t:98]: Maybe try to remove the `--dry-run` argument!

Comment (gpetiot) [reply tests/bin/x-commit-hash/run.t]: good catch

Comment (gpetiot) [reply tests/bin/x-commit-hash/run.t]: although this field contains the commit id and not the tag if there is one, is this what you had in mind for #222 ?

Comment (NathanReb) [reply tests/bin/x-commit-hash/run.t]: That's what I had in mind yes!

**Comment by kit-ty-kate** [comment]
I wasn't aware of this but it seems rather useful. Thanks!

**Pull request merged**